query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -568,4 +570,30 @@ private void validateQueryRequest(ExecutionContext executionContext, QueryReques
           "If distinct selections are requested, there should be no groupBys or aggregations.");
     }
   }
+
+  @VisibleForTesting
```

Comment:
Please add a comment saying this is a temporary fix and also mention the pinot bug here.

Reply:
Done